dulwich/refs.py

```diff
@@ -687,6 +688,14 @@ def set_if_equals(self, name, old_ref, new_ref, committer=None,
         except (KeyError, IndexError):
             realname = name
         filename = self.refpath(realname)
+
+        # make sure none of the ancestor folders is an existing ref
+        probe_ref = os.path.dirname(realname)
+        while probe_ref:
+            if self.read_ref(probe_ref) is not None:
+                raise GitLockingError("cannot lock '%s': '%s' exists" % (realname, probe_ref))
```

Comment:
This code was not attempting to lock, so it's misleading to raise a locking error.